CONTRIBUTING.md

```diff
@@ -62,6 +62,24 @@ When a cluster is first created with a cluster config file, there is no master n
 * The machine controller creates resources (Machines etc)
 * Pivot the apiserver and the machine controller in to the cluster.
 
+#### Configurable Machine Setup
+
+While not mandatory, it is suggested for new providers to support configurable machine setups for creating new machines. 
+This is to allow flexibility in what startup scripts are used and what versions are supported instead of hardcoding startup scripts into the machine controller.
+A yaml file (e.g. `machine_setup_configs.yaml`) should hold the list of valid machine setup configs, and the yaml file will be volume mounted into the machine controller using a ConfigMap named `machine-setup`. 
```

Comment:
It may help to structure it more like:
While not mandatory ....
For GCE, A yaml file ....

That way you are clear about the transition from talking about suggestion for other providers to talking about what is implemented for GCE. Readers will be less confused on what bit is suggestion and what bit is just explaining the current code. Eg. Someone might otherwise wonder if we are saying the ConfigMap needs to be named 'machine-setup' etc.

Reply:
Yeah that makes sense. I made a subheader for GCE Implementation to specifically differentiate the intention from implementation and bumped some stuff around.